env_spec.py

```diff
@@ -19,138 +24,158 @@
 def create_list(input_str):
     """
     Creates a list from input_str_list.Returns the new list.
+    If input_str=[1,2,], returns empty list(syntax error).
     """
     input_str = input_str.replace("[", "")
     input_str = input_str.replace("]", "")
     new_list = input_str.split(",")
 
+    if "" in new_list:
+        return []
+
     for pos in range(0, len(new_list)):
-        new_list[pos] = str(new_list[pos])
+        new_list[pos] = str(new_list[pos]).strip()
 
     return new_list
 
 
-def check_input(input_field):
-    """
-    Checks syntax of input field(uppercase latin chars,numeric digits, underscores).
-    Returns False for error or True for correct syntax.
-    """
-    alphanumeric_that_does_not_start_with_digit = r"^[A-Z_][0-9A-Z_]*$"
-
-    ret = re.match(alphanumeric_that_does_not_start_with_digit, input_field)
-
-    if ret is None:
-        return False
-
-    return True
-
-
-def check_type(input_type):
-    """
-    Checks if given type is valid and returns True, else returns False.
-    """
+def render_to_html(env_spec_list):
+    html_output = ""
 
-    if "[" in input_type and "=" in input_type:
-        env_var_type, value = input_type.split("=")
-        value = value.strip()
+    for dict in env_spec_list:
+        if dict["choices"] is None:
+            if dict["default_value"] is None:
+                ret_str = (
+                    f'<label for="env_spec_{dict["name"].lower()}">{dict["name"]}</label>\n'
+                    f'<input id="env_spec_{dict["name"].lower()}" name="{dict["name"].lower()}" type="{dict["type"]}" />\n'
+                )
+            elif dict["default_value"] is not None:
+                ret_str = (
+                    f'<label for="env_spec_{dict["name"].lower()}">{dict["name"]}</label>\n'
+                    f'<input id="env_spec_{dict["name"].lower()}" name="{dict["name"].lower()}" type="{dict["type"]}" value="{dict["default_value"]}" />\n'
+                )
+        elif dict["choices"] is not None:
+            ret_str = (
+                f'<label for="env_spec_{dict["name"].lower()}">{dict["name"]}</label>\n'
+                f'<select id="env_spec_{dict["name"].lower()}" name="{dict["name"].lower()}">\n'
+            )
 
-        if value in env_var_type:
-            return True
-        else:
-            return False
-    if "=" in input_type:
-        return True
+            for x in range(0, len(dict["choices"])):
+                if (
+                    dict["default_value"] is not None
+                    and dict["default_value"] == dict["choices"][x]
+                ):
+                    ret_str += f'\t<option value="{dict["choices"][x]}"selected>{dict["choices"][x]}</option>\n'
```

Comment:
I would move `option` rendering into a new function, since output is too similar:

Since you are generating `input` elements in multiple places, why not move this into a function to avoid code duplication?

```py
def render_choice(choice, selected=False):
    # Do your things here...

# ...

# Use it like that
render_choice(choice, choice == default_value)
```